src/Utility/External/SortedList.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * SortedList Object. Sorted Array of element (ascending or descending order)
+ *
+ * @author KhaaZ
+ *
+ * @template T
+ *
+ * @class SortedList
+ * @prop {Boolean} acending
+ * @prop {(toInsert, baseElement) => Boolean)} comparator - Whether toInsert is bigger than baseElement
+ * @prop {Array<T>} list
+ */
+class SortedList {
+    constructor(comparator = null, ascending = true) {
+        console.log(comparator);
+        this.comparator = comparator || ( (toInsert, baseElement) => toInsert >= baseElement);
+        this.ascending = ascending;
+        this.list = [];
+    }
+
+    /**
+     * Size of the list
+     *
+     * @readonly
+     * @typedef {Number}
+     * @memberof SortedList
+     */
+    get size() {
+        return this.list.length;
+    }
+
+    /**
+     * Whether the List is empty or not
+     *
+     * @returns {Boolean}
+     * @memberof SortedList
+     */
+    isEmpty() {
+        return this.list.length === 0;
+    }
+
+    /**
+     * Returns the first element of the List without removing it
+     *
+     * @returns {T}
+     * @memberof SortedList
+     */
+    first() {
+        return this.list[0];
+    }
+
+    /**
+     * Returns the first element of the List without removing it
```

Comment:
```suggestion
     * Returns the last element of the list without removing it
```
Also lowercase every `List` please, consistency